### PR TITLE
Assignment5Classes 1

### PR DESCRIPTION
**Pull request by froschs**
First publish of the Drive program with the Car class and Store program
with the RetailItem class

### CODE REVIEW

File: Assignment5Classes/src/carclasses/Car.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+     */
+    public int getYearModel(){ return this.yearModel; }
+    public int getSpeed(){ return this.speed; }
+    public String getMake(){ return this.make; }
+    
+    /**
+     * Public API to increase the speed of the car
+     * @param accelerate Will increase speed by 5 
+     */
+    public void accelerate(int speed){
```
Comment (ninjapanzer) [inline Assignment5Classes/src/carclasses/Car.java]: Do you ever use this argument in your accelerate method?

Comment (ninjapanzer) [inline Assignment5Classes/src/carclasses/Car.java]: This is a good separation of command and query. There was the option to make this return the new speed after mutation. :+1:

File: Assignment5Classes/src/retailclasses/Store.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        int unitsOnHand;
+        double price;
+        Scanner keyboard = new Scanner(System.in);
+        
+        /**
+         * Outputs prompts for user to enter description, amount on hand, and
+         * price for item 1
+         */
+        System.out.println("Enter the description for item 1:");
+        description = keyboard.nextLine();
```
Comment (ninjapanzer) [inline Assignment5Classes/src/retailclasses/Store.java:36]: Continuous reassignment of these variables is a little risky. There is a principle called DRY (don't repeat yourself). These sections are a prime example of where some abstraction can clean up these repeated sections and assure that the logic is contained and reduce side effects.

**Comment by ninjapanzer** [comment]
You should review the requirements for question 4 because you are missing some of the requested methods.

File: Assignment5Classes/src/retailclasses/Store.java (hunk 2 above)
Comment (srtinkey) [inline Assignment5Classes/src/retailclasses/Store.java:36]: @ninjapanzer, I'm not sure I follow you here. I did pretty much the same thing in my program. I see no reason to create additional variables every time I want to create a new item object especially if the fields are the same for each object.

**Comment by srtinkey** [comment]
Some of your comments are confusing. You don't have to use Javadoc formatting for everything. @param is used only when describing parameters for methods.

File: Assignment5Classes/src/retailclasses/Store.java (hunk 2 above)
Comment (ninjapanzer) [inline Assignment5Classes/src/retailclasses/Store.java:36]: @srtinkey There are two reasons to create variables as close to usage as possible:
1. In the event of a large driver(main file) it makes the visible read distance between the declaration and usage of a variable much smaller(don't have to scroll). I am sure at some point you may have been taught to put all your declarations at the top of your file. This is an old behavior. They should be listed at the top of your usage block.
2. To make it easier to extract logic into methods. If we look at each block of assignment and scanner reading they are effectively the same. The DRY rule says "Don't repeat yourself" these blocks would be considered repetitive and a single method definition could encapsulate this logic. It could take scanner as an argument and perform the read and assign variables finally returning a new `RetailItem` instance. In each method body as a result would create its own variables for `description`, `unitsOnHand`, and `price` on each invocation.

If each of the declarations was near the logic group you would see that it is very repetitive and that would directly influence you to extract a method.

Java has garbage collection(GC) for variables as well. This means it returns allocated memory(variable space) to the computer after a variable falls out of scope. In other languages like C++ GC is something new and isn't always taught so using the same variable over and over keeps you from wasting heap space(memory) because you forgot to deallocate(manually give the memory back) the memory.

GC runs on an interval but it is easier for the JVM to know what is out of scope when the logic is encapsulated in a method. This means that after the method is invoked 3 times the original variable space used by `description`, `unitsOnHand`, and `price` will be returned to the system and all that will remain is the storage used for the `RetailItem` object.

Especially since primitives are pass by value we get a new copy in memory for the Object instance as well as what is defined in the driver.

This rule is a little less effective when your object has fields to a bunch of other objects but the GC still works the same.

TLDR;
You should consider putting all of the logic in your driver in methods and the bulk of your driver will be control statements(if, loops) and method invocations.

**Comment by froschs** [comment]
The methods for question 4 I was missing were the mutators, right? I included those in the most recent commit.
